docs/api-reference/geo-layers/tile-layer.md

```diff
@@ -307,8 +310,43 @@ Properties:
 - `isVisible` (Boolean) - if the tile should be rendered
 - `isLoaded` (Boolean) - if the content of the tile has been loaded
 
-## Source
+## Tileset2D
 
-[modules/geo-layers/src/tile-layer](https://github.com/visgl/deck.gl/tree/master/modules/geo-layers/src/tile-layer)
+Class that manages loading and purging of tile data. This class caches recently visited tiles and only creates new tiles if they are present.
 
+To implement a custom indexing scheme, extend `Tileset2D` and implement the following interface:
 
+- `getTileIndices({viewport, maxZoom, minZoom, zRange, modelMatrix, modelMatrixInverse})` - returns an array of indices in the given viewport. The indices should have the shape of Objects, like: `{q: '0123'}` to allow referencing in the URL template supplied to the `data` prop.
+- `getTileId(index)` - returns unique string key for a tile index.
+- `getParentIndex(index)` - returns index of the parent tile.
+- `getTileZoom(index)` - returns a zoom level for a tile index.
+- `getTileMetadata(index) - returns additional metadata to add to tile (optional).
+
+For example, to index using quadkeys:
+
+```js
+class QuadkeyTileset2D extends Tileset2D {
+  getTileIndices(opts) {
+    // Quadkeys and OSM tiles share the layout, leverage existing algorithm
+    // Data format: [{q: '0120'}, {q: '0121'}, {q: '0120'}, {q: '0120'},...]
+    return super.getTileIndices(opts).map(tileToQuadkey);
+  }
+
+  getTileId({q}) {
+    return q;
+  }
+
+  getTileZoom({q}) {
+    return q.length;
+  }
+
+  getParentIndex(index) {
+    index.q = index.q.slice(0, -1);
+    return index;
+  }
```

Comment:
Do you think it would be helpful to add short one-line explanations of how this works in the context of quadkeys? Or should we expect that readers would be familiar with how quadkeys work?

Reply:
I would rather avoid getting into the details as the whole point of exposing the `Tileset2D` class is to allow an arbitrary indexing scheme and so it is sufficient to state what interface must be implemented without specific details on the example. That said, I'll add a link to an article on quadkeys to help readers